test/templateupdaterequest/templateupdaterequest.go

```diff
@@ -82,3 +121,24 @@ func (t TierName) applyToTemplateUpdateRequest(r *toolchainv1alpha1.TemplateUpda
 		toolchainv1alpha1.NSTemplateTierNameLabelKey: string(t),
 	}
 }
+
+// SyncIndexes the Sync Indexes stored in the status before the MasterUserRecord update begins
+type SyncIndexes map[string]string
+
+var _ Option = SyncIndexes(map[string]string{})
+
+func (i SyncIndexes) applyToTemplateUpdateRequest(r *toolchainv1alpha1.TemplateUpdateRequest) {
+	r.Status.SyncIndexes = map[string]string(i)
+}
+
+// Condition adds a condition in the TemplateUpdateRequest status
+type Condition toolchainv1alpha1.Condition
+
+var _ Option = Condition(toolchainv1alpha1.Condition{})
+
+func (c Condition) applyToTemplateUpdateRequest(r *toolchainv1alpha1.TemplateUpdateRequest) {
+	if r.Status.Conditions == nil {
+		r.Status.Conditions = []toolchainv1alpha1.Condition{}
+	}
+	r.Status.Conditions = append(r.Status.Conditions, toolchainv1alpha1.Condition(c))
```

Comment:
append accepts nil as the first argument 
```suggestion
	r.Status.Conditions = append(r.Status.Conditions, toolchainv1alpha1.Condition(c))
```

Reply:
oh nice, I was not aware about this! Cool!